### ISSUE TITLE
Update CMake Documentation

### ISSUE THREAD

**Issue by Lnk2past** [conan-io/docs#695]
Looking [here](https://docs.conan.io/en/latest/reference/build_helpers/cmake.html)

This documentation is not up-to-date when compared to the latest code. Namely, I was looking at the documentation for the [configure method](https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#configure).

The code uses the following signature:
```
def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
              source_folder=None, build_folder=None, cache_build_folder=None,
              pkg_config_paths=None):
```

The documentation specifies the signature as:

```
def configure(self, args=None, defs=None, source_folder=None, build_folder=None,
              cache_build_folder=None, pkg_config_paths=None)
```

This should be updated. Sadly I am prohibited from making a PR from work, if no one else wants to address it, perhaps I can try to do so from home.

**Comment by Lnk2past** [comment]
Also, I do understand that the missing params here are deprecated, but that should be indicated in the documentation.

**Comment by Lnk2past** [comment]
@Croydon why? The documentation still references the preferred signature and the deprecated parameters are still available. This would be OK to close if one of the following happens:

1. The deprecated parameters and any logic used to differentiate them from the preferred ones are removed.

OR

2. Conan produces a warning when the deprecated parameters are used.

OR

3. The documentation is updated to specify their availability and deprecation.

I have listed these in order by what I think would be best; removing them would be best and if not warn on usage. Otherwise just document them.

**Comment by danimtb** [comment]
This was partially accomplished by #960.

I am creating a PR right now to have it completely fixed

**Comment by danimtb** [comment]
fixed in #992
